### PR TITLE
Form

### PR DESCRIPTION
**Pull request by maxmodlol**

### CODE REVIEW

File: form.html [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<!DOCTYPE html>
+<html>
+   
+<body>
+
+<h2>HTML Forms</h2>
+
+<form action="/action_page.php">
+  <label for="fname">First name:</label><br>
+  <input type="text" id="fname" name="fname" pattern=".{3,}"  ><br>
```
Comment (Abdallah-Via) [inline form.html]: - don't forget to close the input tag .../>
- don't use <br>
- Every element (label + input) put them in a div

File: form.html [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  <input type="email" id="email" > <br>
+  <label for="color">Favorite Color</label><br>
+  <input type="color" id="color" > <br>
+  <label for="image">Favorite Color</label><br>
+  <input type="file" id="image" accept="image/png,image/gif,image/jpeg" ><br>
+  <p>Please select your gender</p>
+    <input type="radio" id="Male" name="gender"  value="male">
+ <label for="Male">Male</label><br>
+ <input type="radio" id="Female" name="gender" value="female">
+<label for="Female">Female</label><br>
```
Comment (Abdallah-Via) [inline form.html]: should be in div

File: form.html [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+<p>Please select your courses</p>
+
+<input type="checkbox" id="java"  value="Java">
+<label for="java"> Java</label><br>
+<input type="checkbox" id="C++" value="C++">
+<label for="C++"> C++</label><br>
+<input type="checkbox" id="python" value="Python">
+<label for="python">Python</label><br>
+<input type="checkbox" id="PHP" value="PHP">
+<label for="PHP">PHP</label><br>
```
Comment (Abdallah-Via) [inline form.html]: should be in a div

File: form.html [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+<label for="C++"> C++</label><br>
+<input type="checkbox" id="python" value="Python">
+<label for="python">Python</label><br>
+<input type="checkbox" id="PHP" value="PHP">
+<label for="PHP">PHP</label><br>  
+<textarea rows = "3"  id = "description"> </textarea><br>
+ <label for="description">Description</label><br>
+
+   
+  <input type="submit" value="Submit">
```
Comment (Abdallah-Via) [inline form.html:58]: on submit, print an object that contains all data from the form

File: form.html [hunk 5]
```diff
@@ -0,0 +1,53 @@
+<!DOCTYPE html>
+<html>
+   
+<body>
+
+<h2>HTML Forms</h2>
+
+<form action="/action_page.php">
```
Comment (Abdallah-Via) [inline form.html]: please check the whole file indentation

File: form.html [hunk 6]
```diff
@@ -1,53 +1,77 @@
...
-  <input type="email" id="email" > <br>
-  <label for="color">Favorite Color</label><br>
-  <input type="color" id="color" > <br>
-  <label for="image">Favorite Color</label><br>
-  <input type="file" id="image" accept="image/png,image/gif,image/jpeg" ><br>
-  <p>Please select your gender</p>
+<form id="my-form" onsubmit="myFunction()">
+  <div>
+    <label for="fname">First name:</label><br>
+    <input type="text" id="fname" name="fname" pattern=".{3,}"><br>
```
Comment (Abdallah-Via) [inline form.html]: it is more readable to use minlength attribute, use pattern in case of you want to apply regex

File: form.html [hunk 7]
```diff
@@ -1,53 +1,77 @@
...
+  </div>
+  <div>
+    <label for="lname">Last name:</label><br>
+    <input type="text" id="lname" name="lname" pattern=".{3,}"><br>
```
Comment (Abdallah-Via) [inline form.html]: same as above

File: form.html [hunk 8]
```diff
@@ -1,53 +1,77 @@
...
+  </div>
+  <div>
+    <label for="date">Date :</label><br>
+    <input type="date" id="date" name="date"  ><br>
```
Comment (Abdallah-Via) [inline form.html]: I want the validation to be directly in the html, you can use min/max attributes

File: form.html [hunk 9]
```diff
@@ -1,53 +1,77 @@
...
+    <label for="salary">Salary:</label><br>
+    <input type="number" id="salary" min="0"> <br>
+  </div>
+  <div>
+    <label for="email">email</label><br>
+    <input type="email" id="email" > <br>
+  </div>
+  <div>
+    <label for="color">Favorite Color</label><br>
+    <input type="color" id="color" > <br>
```
Comment (Abdallah-Via) [inline form.html]: make the default color green

File: form.html [hunk 10]
```diff
@@ -1,53 +1,77 @@
...
+  </div>
+  <div>
+    <label for="image">Favorite Color</label><br>
```
Comment (Abdallah-Via) [inline form.html]: rename the label

File: form.html (hunk 8 above)
Comment (maxmodlol) [reply form.html]: but how im going to get the date now in html only ?

**Review by Abdallah-Via — APPROVED** [review]

**Pull request merged**